src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeysTableTitle.vue

```diff
@@ -0,0 +1,156 @@
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+ -->
+<template>
+  <div class="table-title" :class="{ 'table-closed': !arrowState }">
+    <div class="title-text" @click="arrowClick">
+      <v-btn icon :color="colors.WarmGrey100">
+        <v-icon v-if="arrowState">icon-Sorting-arrow</v-icon>
+        <v-icon v-else class="arrow-degree">icon-Sorting-arrow</v-icon>
+      </v-btn>
+
+      {{ $t(title) }}
+    </div>
+    <div class="status-wrap">
+      <div v-if="errors > 0" class="errors">
+        <v-icon color="red">icon-Error</v-icon> {{ errorCount }}
+        {{ $t('keys.globalErrors') }}
+      </div>
+      <div v-else-if="registered === certificateCount" class="registered">
+        <v-icon color="green">icon-Checked</v-icon> {{ $t('keys.noIssues') }}
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+// View for a token
+import Vue from 'vue';
+import { Colors } from '@/global';
+import { CertificateStatus, Key, TokenCertificate } from '@/openapi-types';
+import { Prop } from 'vue/types/options';
+
+export default Vue.extend({
+  props: {
+    keys: {
+      type: Array as Prop<Key[]>,
+      required: true,
+    },
+    title: {
+      type: String,
+      required: true,
+    },
+    arrowState: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  computed: {
+    errorCount(): number {
+      let errors = 0;
+
+      this.keys.forEach((key: Key) => {
+        key.certificates.forEach((cert: TokenCertificate) => {
+          if (cert.status === CertificateStatus.GLOBAL_ERROR) {
+            errors++;
+          }
+        });
+      });
+
+      return errors;
+    },
+  },
+  data() {
+    return {
+      errors: 0,
+      registered: 0,
+      certificateCount: 0,
+      colors: Colors,
+    };
+  },
+  methods: {
+    countStates(): void {
+      this.keys.forEach((key: Key) => {
+        key.certificates.forEach((cert: TokenCertificate) => {
+          if (cert.status === CertificateStatus.GLOBAL_ERROR) {
+            this.errors++;
+          } else if (cert.status === CertificateStatus.REGISTERED) {
+            this.registered++;
+          }
+          this.certificateCount++;
+        });
+      });
```

Comment:
flatMap could be utilised here aswell but you may decide whether you want to refactor it.

Reply:
Fixed